src/hooks/useDataSource.ts

```diff
@@ -106,7 +110,9 @@ async function importBook(storage: BookStorage | null, rawData: Uint8Array) {
   const rawBook = await importGnucashBook(parsedData);
   await DATASOURCE.sqljsManager.loadDatabase(rawBook);
 
+  // Remove all previous data from the state
   mutate((key: string) => key.startsWith('/api'), undefined);
+  await preloadData();
```

Comment:
When importing the book, we also want to preload the data right after purging old data.